internal/utils/exec.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+func Exec(log func(format string, a ...interface{}), command string, args ...string) ([]byte, error) {
+	c := exec.Command(command, args...)
+	bytes, err := c.CombinedOutput()
+	if err != nil {
+		cmdStr := fmt.Sprintf("%s %s", command, strings.Join(args, " "))
+		log("❌ Unable to execute %q:", cmdStr)
```

Comment:
Why not just return error here? Haven't seen this pattern before. This also makes the assumption the caller always wants a log message with the ❌ prefix.

Reply:
We'd have to log the output in all of the callers then. I think in all of the current call sites it's beneficial logging the stdout/stderr in case the command execution failed. Thus, I also think it makes sense to keep the logging in this utility func, even if it's a rather unconventional pattern.